### PR TITLE
cargo_fetch PG, rust, cargo: Add support for using legacysupport PG 

### PR DESCRIPTION
**Pull request by cjones051073**
https://trac.macports.org/ticket/62656

Support older systems requiring legacy support 

    - Patch bootstrapping binaries to use legacysupport dylib for missing libSystem calls
    - Wrap underlying compiler commands used by rust/cargo to enforce use of legacy support flags

Fixes things perfectly for macOS10.9. rust, cargo install fine and was able to build a number of cargo/rust based ports.

Work on older systems is WIP, as currently failing for other unknown reasons.

@kencu FYI

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by herbygillot** [comment]
I've gotta be honest, I'm still pretty wary of doing any `legacysupport` modifications to anything Rust-related.

File: lang/rust/Portfile [hunk 1]
```diff
@@ -2,6 +2,11 @@
...
 PortSystem          1.0
 
+# Needed to patch the bootstrap binaries and for the final build
+PortGroup                     legacysupport 1.1
+legacysupport.use_static                    yes
+legacysupport.newest_darwin_requires_legacy 13
+
 name                rust
 version             1.51.0
 revision            1
```
Comment (herbygillot) [inline lang/rust/Portfile:12]: Shouldn't Rust's revision be bumped for all these changes?

**Comment by cjones051073** [comment]
> I've gotta be honest, I'm still pretty wary of doing any `legacysupport` modifications to anything Rust-related.

I understand, there was problems last time. I know why there was problems last time this was tried, and this time I avoid them. For one, last time the use of legacysupport was not limited to 10.9 or older, so it was also used on 10.10 and 10.11. This time it is currently limited to 10.9 and older, so will only affect platforms rust currently does not build on anyway. 

The other reason it shouldn’t be a problem this time is i use the static link option, which avoids the issue with the internal llvm build, and also wrap the commands to make completely sure the right flags are used.

File: lang/rust/Portfile (hunk 1 above)
Comment (cjones051073) [reply lang/rust/Portfile:12]: No. Nothing is changed at all for 10.10 and new, and for 10.9 and older rust doesn’t currently build.

**Comment by cjones051073** [comment]
Tested now extensively on a number of systems. Fixes build of rust itself on 10.9, and also on 10.{10-11} a number of port builds that currently fail due to missing features in libSystem, like clock_gettime.

**Pull request merged**